brave/brave-core
# Sidebar Slide animation

fix https://github.com/brave/brave-browser/issues/21992
fix https://github.com/brave/brave-browser/issues/25382


https://github.com/brave/brave-core/assets/6786187/8d518eee-c0bb-4e01-96bb-140d3bb58ca9


https://github.com/brave/brave-core/assets/6786187/98d79b77-a308-4891-9a67-ba2c93f6d8ae


https://github.com/brave/brave-core/assets/6786187/220eb662-51a4-4845-b22d-485889edb552

Added slide animation for opening/closing sidebar/side panel.
To use slide animation, this PR has some more refactoring/cleanup.

1. Ask to show/hide sidebar/panel when only needed.
2. Use SidePanelUI(SidePanelCoordinator) for starting side panel activating.
     So far, we use both(SidePanelUI/SidebarController) for activating side panel item and
     this causes some methods are called multiple times for same operation request.
3. Renamed ambiguous`Sidebar::UpdateSidebar()` to `Sidebar::UpdateSidebarItemsState()` as
    it's called only when item's status should be updated.
4. Removed `SidebarController::UpdateSidebarVisibility()`. Instead of asking sidebar visibility update,
    `SidebarContainerView` updates its child view's visibility via active index change notification.
     

Resolves 



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

`npm run test brave_browser_tests -- --filter=*Sideabar*`

1. Launch browser and do open/closing sidebar on any sidebar show options
2. Check animation is done properly
